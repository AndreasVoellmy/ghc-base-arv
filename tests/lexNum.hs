module Main where

testStrings 
 = [	"0x3y",
     	"0X3abx",
     	"0o39y",
     	"0O334z",
   
     	"NaN", 
     	"Infinity",
   
     	"34yy",
    	"34.4x",
    	"034.4x",
	"31.45e-6y",
	"49.2v",
	"049.2v",
	"35e-3x",
	"035e-3x",
	"35e+3y",
	"83.3e-22",
	"083.3e-22"
   ]

main = mapM test testStrings

test s = print (lex s)

