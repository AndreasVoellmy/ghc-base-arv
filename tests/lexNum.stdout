[("0x3","y")]
[("0X3ab","x")]
[("0o3","9y")]
[("0O334","z")]
[("NaN","")]
[("Infinity","")]
[("34","yy")]
[("34.4","x")]
[("034.4","x")]
[("31.45e-6","y")]
[("49.2","v")]
[("049.2","v")]
[("35e-3","x")]
[("035e-3","x")]
[("35e+3","y")]
[("83.3e-22","")]
[("083.3e-22","")]
